src/protocols/protocol_address_31402.cpp

```diff
@@ -102,11 +102,13 @@ bool protocol_address_31402::handle_receive_get_address(const code& ec,
     if (stopped(ec))
         return false;
 
-    // TODO: allowing repeated queries can allow a channel to map our history.
-    // TODO: pull active hosts from host cache (currently just resending self).
-    // TODO: need to distort for privacy, don't send currently-connected peers.
-    // TODO: response size limit is max_address (1000).
+    bc::message::network_address::list addresses;
+    network_.fetch_addresses(addresses);
 
+    static auto generator = std::default_random_engine{pseudo_random()};
+    std::shuffle(addresses.begin(), addresses.end(), generator);
```

Comment:
I think it would be more meaningful obscuration to select from the full list in a randomized manner, as opposed to taking a fixed slice of the full list from a random offset and then reordering it.